### PR TITLE
Navigation Editor (Storybook)

### PR DESCRIPTION
**Pull request by sagaidak**

### CODE REVIEW

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+type NavTabContent = FilterByTags | SpecificPost | OuterUrl
+
+export interface NavTab {
+  id: number
+  title: string
+  content: NavTabContent
+  description: string
+  hidden: boolean
+  type: string
```
Comment (mell-old) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]: Maybe create type?

**Comment by mell-old** [comment]
Use eslint for all file!!!

File: src/components/stories/navigation-editor/NavigationEditor.css [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+.NavigationEditor .NEErrorMessage {
+  position: absolute!important;
+  top: 5.5rem;
+  left: 6rem;
+}
+
+.NavigationEditor .NEAutoCompleteItem {
+  position: absolute;
+
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.css]:
```suggestion
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+          Save
+        </Button>
+      </Form>
+
+    </Section>
+  </>
+}
+
+// Validation
+const TITLE_REGEX = /^[A-Za-z0-9_-]+$/;
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]: Title could be any text
```suggestion
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+const TITLE_MIN_LEN = 2;
+const TITLE_MAX_LEN = 50;
+
+const schema = Yup.object().shape({
+  navTabs: Yup.array()
+    .of(
+      Yup.object().shape({
+        title: Yup.string()
+          .matches(TITLE_REGEX, 'Title can have only letters (a-z, A-Z), numbers (0-9), underscores (_) and dashes (-).')
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]:
```suggestion
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+          .min(TITLE_MIN_LEN, 'Title is too short (min length: 2)')
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]:
```suggestion
          .min(TITLE_MIN_LEN, `Title is too short. Min length is ${TITLE_MIN_LEN} chars.`)
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+          .max(TITLE_MAX_LEN, 'Title is too long (max length: 50)')
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]:
```suggestion
          .max(TITLE_MAX_LEN, `Title is too long. Max length is ${TITLE_MAX_LEN} chars.`)
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+          .required('Required message')
+      })
+    )
+});
+
+// The type of props MyForm receives
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]:
```suggestion
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+
+const schema = Yup.object().shape({
+  navTabs: Yup.array()
+    .of(
+      Yup.object().shape({
+        title: Yup.string()
+          .matches(TITLE_REGEX, 'Title can have only letters (a-z, A-Z), numbers (0-9), underscores (_) and dashes (-).')
+          .min(TITLE_MIN_LEN, 'Title is too short (min length: 2)')
+          .max(TITLE_MAX_LEN, 'Title is too long (max length: 50)')
+          .required('Required message')
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]:
```suggestion
          .required('This field is required')
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+  mapPropsToValues: props => {
+    return {
+      navTabs: props.navTabs
+    };
+  },
+
+  validationSchema: schema,
+
+  handleSubmit: values => {
+    console.log(values)
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]:
```suggestion
    // console.log(values)
```

File: src/components/stories/navigation-editor/NavigationEditor.tsx [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+                  onClick={() => { arrayHelpers.push(defaultTab) }}
+                >
+                  + Add Tab
+                </div>
+              </div>
+            </div>
+          )}
+        />
+
+        <Button type="primary" htmlType="submit" disabled={false} className={'NESubmit'}>
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.tsx]:
```suggestion
        <Button type='primary' htmlType='submit' disabled={!isValid || isSubmitting} className={'NESubmit'}>
```

File: src/components/stories/tags-input/TagsInput.tsx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+      )}
+      
+      {!inputVisible && (
+        <Tag onClick={showInput} style={{ background: '#fff', borderStyle: 'dashed' }}>
+          <Icon type="plus" /> New Tag
+          </Tag>
+      )}
+    </div>
+  );
+
```
Comment (siman) [inline src/components/stories/tags-input/TagsInput.tsx]:
```suggestion
```

File: src/components/stories/tags-input/TagsInput.tsx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+          }
+        >
+          <Input onPressEnter={handleInputConfirm}  />
+        </AutoComplete>
+      )}
+      
+      {!inputVisible && (
+        <Tag onClick={showInput} style={{ background: '#fff', borderStyle: 'dashed' }}>
+          <Icon type="plus" /> New Tag
+          </Tag>
```
Comment (siman) [inline src/components/stories/tags-input/TagsInput.tsx]:
```suggestion
        </Tag>
```

File: src/components/stories/navigation-editor/NavigationEditor.css [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+}
+.NavigationEditor .NERow:last-child {
+  border-bottom: none;
+}
+
+.NavigationEditor .NEText {
+  color       : #000;
+  font-style  : normal;
+  font-weight : 500;
+  font-size   : 1rem;
```
Comment (siman) [inline src/components/stories/navigation-editor/NavigationEditor.css]: Use our CSS vars here for the font size and margins

File: src/components/stories/tags-input/TagsInput.tsx [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+          style={{ width: 78 }}
+          onChange={handleInputChange}
+          onBlur={handleInputConfirm}
+          value={inputValue}
+          dataSource={tagsData}
+          filterOption={(inputValue, option) =>
+            option.props.children?.toString().toUpperCase().indexOf(inputValue.toUpperCase()) !== -1
+          }
+        >
+          <Input onPressEnter={handleInputConfirm}  />
```
Comment (siman) [inline src/components/stories/tags-input/TagsInput.tsx]:
```suggestion
          <Input onPressEnter={handleInputConfirm} />
```

**Pull request closed without merging**